### PR TITLE
Added new query and criteria for correct removing of older backups

### PR DESCRIPTION
**Pull request by ToshY**
Issue #63

### CODE REVIEW

**Comment by ToshY** [comment]
@bastien70 

FYI according to the contributing guidelines, I'm supposed to ran the CI/test jobs locally. CI was fine, but the tests failed due to "Too Many Connections" related to the database. 

```
<!-- An exception occurred in the driver: SQLSTATE[08004] [1040] Too many connections (500 Internal Server Error) -->'
    'version' => '1.1'
    'statusCode' => 500
    'statusText' => 'Internal Server Error'
    'charset' => 'UTF-8'
) is redirected and has header "Location" with value "/login".

/app/vendor/symfony/framework-bundle/Test/BrowserKitAssertionsTrait.php:142
/app/vendor/symfony/framework-bundle/Test/BrowserKitAssertionsTrait.php:56
/app/tests/Controller/SecurityControllerTest.php:33
/app/tests/Controller/SecurityControllerTest.php:15

Caused by
ErrorException: An exception occurred in the driver: SQLSTATE[08004] [1040] Too many connections in /app/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:119
Stack trace:
#0 /app/vendor/symfony/framework-bundle/Test/BrowserKitAssertionsTrait.php(56): Symfony\Bundle\FrameworkBundle\Test\WebTestCase::assertThatForResponse(Object(PHPUnit\Framework\Constraint\LogicalAnd), '')
#1 /app/tests/Controller/SecurityControllerTest.php(33): Symfony\Bundle\FrameworkBundle\Test\WebTestCase::assertResponseRedirects('/login')#2 /app/tests/Controller/SecurityControllerTest.php(15): App\Tests\Controller\SecurityControllerTest->assertFailedLogin('user100@test.co...', 'test')
#3 /app/vendor/phpunit/phpunit/src/Framework/TestCase.php(1545): App\Tests\Controller\SecurityControllerTest->testLogin()
#4 /app/vendor/phpunit/phpunit/src/Framework/TestCase.php(1151): PHPUnit\Framework\TestCase->runTest()
#5 /app/vendor/phpunit/phpunit/src/Framework/TestResult.php(726): PHPUnit\Framework\TestCase->runBare()
#6 /app/vendor/phpunit/phpunit/src/Framework/TestCase.php(903): PHPUnit\Framework\TestResult->run(Object(App\Tests\Controller\SecurityControllerTest))
#7 /app/vendor/phpunit/phpunit/src/Framework/TestSuite.php(677): PHPUnit\Framework\TestCase->run(Object(PHPUnit\Framework\TestResult))
#8 /app/vendor/phpunit/phpunit/src/Framework/TestSuite.php(677): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#9 /app/vendor/phpunit/phpunit/src/Framework/TestSuite.php(677): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#10 /app/vendor/phpunit/phpunit/src/TextUI/TestRunner.php(673): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#11 /app/vendor/phpunit/phpunit/src/TextUI/Command.php(143): PHPUnit\TextUI\TestRunner->run(Object(PHPUnit\Framework\TestSuite), Array, Array, true)
#12 /app/vendor/phpunit/phpunit/src/TextUI/Command.php(96): PHPUnit\TextUI\Command->run(Array, true)
#13 /app/vendor/phpunit/phpunit/phpunit(98): PHPUnit\TextUI\Command::main()
#14 /app/vendor/bin/phpunit(123): include('/app/vendor/php...')
#15 {main}
ERRORS!
```

While this particular issue will probably not occur in the CI (for now), I think this has to do with the fact that I dockerized the app and added the image to another app already (connect to same MySQL service with max 5 active connections, of which only 2 were active without running tests).

So after some searching, I've found that if I added `processIsolation = "true"` to the `phpunit.xml.dist`, it stopped giving these errors (it now just fails on 1 test case unrelated to this change). I'm definitely not that familiar with unit testing, so I have no idea what the impact of `processIsolation` is, but apparently setting it to `true` works. Is this something that could be changed?

**Comment by bastien70** [comment]
Oh sorry I didn't see your comment.

Well I guess that shouldn't be a problem. @jmsche  ?

File: src/Repository/BackupRepository.php [hunk 1]
```diff
@@ -20,4 +21,35 @@ public function __construct(ManagerRegistry $registry)
...
+                $qb->expr()->eq(
+                    'd.id',
+                    $database->getId()
+                )
+            )
+            ->orderBy('b.createdAt', 'DESC')
+            ->setMaxResults($database->getMaxBackups() - 1);
+
+        return $qb->getQuery()->getResult();
+    }
```
Comment (bastien70) [inline src/Repository/BackupRepository.php:54]: Maybe something like : 

```php
return $this->createQueryBuilder('b')
            ->join('b.database', 'd')
            ->andWhere('d.id = :database')
            ->setParameter('database', $database->getId())
            ->orderBy('b.createdAt', 'DESC')
            ->setMaxResults($database->getMaxBackups() - 1)
            ->getQuery()
            ->getResult();
```

Comment (ToshY) [reply src/Repository/BackupRepository.php:54]: Ah yes, definitely forgot `setParameter` 🤦‍♂️. I'm fine with the usage of querybuilder like you stated, I'm just used to the other way. Only thing I would change is `:database` to `:databaseId`, just to clarify it's not the entity but the ID.

**Comment by bastien70** [comment]
Okay for me for the query. For last changes, I let jmsche take care of it

**Review by jmsche — APPROVED** [review]

**Pull request merged**